sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -778,5 +778,11 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def clear(): Unit = {
     settings.clear()
   }
+
+  def reset(): Unit = {
+    sqlConfEntries.values.asScala.foreach { entry =>
```

Comment:
Actually, we allow users set/define the parameters that are not part of `sqlConfEntries`. However, this is not allowed in `Hive`. Thus, I am not 100% sure which is the best behavior for the `reset` command.